htp/Makefile.am

```diff
@@ -27,5 +27,5 @@ libhtp_c_la_SOURCES = $(h_sources) $(h_sources_private) $(c_sources)
 
 lib_LTLIBRARIES = libhtp.la
 libhtp_la_SOURCES =
-libhtp_la_LIBADD = libhtp-c.la lzma/liblzma-c.la
-libhtp_la_LDFLAGS = -version-info $(GENERIC_LIBRARY_VERSION)
+libhtp_la_LIBADD = libhtp-c.la lzma/liblzma-c.la $(LTLIBICONV)
+libhtp_la_LDFLAGS = -version-info $(GENERIC_LIBRARY_VERSION) -no-undefined
```

Comment:
Could you explain why we need this flag `-no-undefined` ?

Reply:
Windows DLLs do not support undefined references.

https://autotools.io/libtool/windows.html
> Because of the explicit nature of the table of imported symbols, it is not possible to leave a symbol in a PE DLL undefined at link time, to be satisfied at runtime, as it is instead possible with most UNIX shared objects.

Reply:
Does it work if you run `LDFLAGS="-no-undefined" ./configure && make` ?

Reply:
It does not work.
```
/bin/sh ../libtool  --tag=CC   --mode=link gcc -I.. -I../htp -D_GNU_SOURCE -g -Wall -Wextra -std=gnu99 -pedantic -Wextra -Wno-missing-field-initializers -Wshadow -Wpointer-arith -Wstrict-prototypes -Wmissing-prototypes -Wno-unused-parameter -O2 -Wstrict-overflow=1 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -fPIC -version-info 2:0:0 -Wl,--no-undefined -o libhtp.la -rpath /usr/local/lib  libhtp-c.la lzma/liblzma-c.la -lz  -liconv
libtool: warning: undefined symbols not allowed in x86_64-pc-cygwin shared libraries; building static only
```